### PR TITLE
Validation - Improvements #37

### PR DESCRIPTION
**Pull request by amamla**
* new json parser
* new structure for template with details regarding original file
  offset

### CODE REVIEW

File: offlinevalidator/parsers/jsonparser.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parsers
+
+import (
+	"github.com/amamla/jsonparser"
```
Comment (ppikula) [inline offlinevalidator/parsers/jsonparser.go]: Maybe, mirror it under Appliscale account

Comment (afronski) [reply offlinevalidator/parsers/jsonparser.go]: :+1: To that.

**Comment by afronski** [comment]
@jlampar Please have a look at that review if we can incorporate @amamla work to the _goformation_ without breaking anything - in order to leverage support for line and column when we have syntax errors.

File: offlinevalidator/parsers/jsonparser.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
+// Copyright 2018 Appliscale
```
Comment (afronski) [inline offlinevalidator/parsers/jsonparser.go]: Please use `go fmt ...` before you commit, rebase it with current `master`.

**Comment by afronski** [comment]
PR for issue #37.

**Pull request merged**